crates/bevy_color/src/srgba.rs

```diff
@@ -0,0 +1,430 @@
+    pub const NONE: Srgba = Srgba::new(0.0, 0.0, 0.0, 0.0);
+    /// <div style="background-color:rgb(50%, 50%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const OLIVE: Srgba = Srgba::new(0.5, 0.5, 0.0, 1.0);
+    /// <div style="background-color:rgb(50%, 0%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const PURPLE: Srgba = Srgba::new(0.5, 0.0, 0.5, 1.0);
+    /// <div style="background-color:rgb(100%, 0%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const RED: Srgba = Srgba::new(1.0, 0.0, 0.0, 1.0);
+    /// <div style="background-color:rgb(75%, 75%, 75%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const SILVER: Srgba = Srgba::new(0.75, 0.75, 0.75, 1.0);
+    /// <div style="background-color:rgb(0%, 50%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const TEAL: Srgba = Srgba::new(0.0, 0.5, 0.5, 1.0);
+    /// <div style="background-color:rgb(100%, 100%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const WHITE: Srgba = Srgba::new(1.0, 1.0, 1.0, 1.0);
+    /// <div style="background-color:rgb(100%, 100%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const YELLOW: Srgba = Srgba::new(1.0, 1.0, 0.0, 1.0);
+
+    /// Construct a new [`Srgba`] color from components.
+    ///
+    /// # Arguments
+    ///
+    /// * `red` - Red channel. [0.0, 1.0]
+    /// * `green` - Green channel. [0.0, 1.0]
+    /// * `blue` - Blue channel. [0.0, 1.0]
+    /// * `alpha` - Alpha channel. [0.0, 1.0]
+    pub const fn new(red: f32, green: f32, blue: f32, alpha: f32) -> Self {
+        Self {
+            red,
+            green,
+            blue,
+            alpha,
+        }
+    }
+
+    /// Construct a new [`Srgba`] color from (r, g, b) components, with the default alpha (1.0).
+    ///
+    /// # Arguments
+    ///
+    /// * `red` - Red channel. [0.0, 1.0]
+    /// * `green` - Green channel. [0.0, 1.0]
+    /// * `blue` - Blue channel. [0.0, 1.0]
+    pub const fn rgb(red: f32, green: f32, blue: f32) -> Self {
+        Self {
+            red,
+            green,
+            blue,
+            alpha: 1.0,
+        }
+    }
+
+    /// New `Srgba` from a CSS-style hexadecimal string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bevy_color::Srgba;
+    /// let color = Srgba::hex("FF00FF").unwrap(); // fuchsia
+    /// let color = Srgba::hex("FF00FF7F").unwrap(); // partially transparent fuchsia
+    ///
+    /// // A standard hex color notation is also available
+    /// assert_eq!(Srgba::hex("#FFFFFF").unwrap(), Srgba::new(1.0, 1.0, 1.0, 1.0));
+    /// ```
+    pub fn hex<T: AsRef<str>>(hex: T) -> Result<Self, HexColorError> {
+        let hex = hex.as_ref();
+        let hex = hex.strip_prefix('#').unwrap_or(hex);
+
+        match *hex.as_bytes() {
+            // RGB
+            [r, g, b] => {
+                let [r, g, b, ..] = decode_hex([r, r, g, g, b, b])?;
```

Comment:
Why not use the built in `from_str_radix` then just do some bitwise operation on either u16 or u32 to convert back to RGBA bytes?

I think using `from_str_radix` and some form of bitwise will be faster due to cache friendness and no loops.

Reply:
Bear in mind that this needs to work with 3, 4, 6 and 8 digits.

I think I will leave this for now. "First make it correct, then make it fast". If someone wants to make an optimized version, well, there's lots of unit tests to validate.